gnomad/utils/annotations.py

```diff
@@ -1519,3 +1529,308 @@ def _create_group_dicts(
+    computed_tmpfile: str = new_temp_file("gks-seqloc-post.tsv"),
+):
+    """
+    Compute sequence location digest-based id for a hail variant table.
+
+    Exports table to tsv, computes SequenceLocation digests, reimports and replaces
+    the vrs_json field with the result. Input table must have a .vrs field, like the
+    one added by add_gks_vrs, that can be used to construct ga4gh.vrs models.
+
+    :param ht: hail table with VRS annotation
+    :param export_tmpfile: file path to export the table to.
+    :param computed_tmpfile: file path to write the updated rows to, which is then imported as a hail table
+    :return: a hail table with the VRS annotation updated with the new SequenceLocations
+    """
+    logger.info("Exporting ht to %s", export_tmpfile)
+    ht.select("vrs_json").export(export_tmpfile, header=True)
+
+    logger.info(
+        "Computing SequenceLocation digests and writing to %s", computed_tmpfile
+    )
+    start = timer()
+    counter = 0
+    with open(computed_tmpfile, "w", encoding="utf-8") as f_out:
+        with open(export_tmpfile, "r", encoding="utf-8") as f:
+            reader = csv.reader(f, delimiter="\t")
+            header = None
+            for line in reader:
+                if header is None:
+                    header = line
+                    f_out.write("\t".join(header))
+                    f_out.write("\n")
+                    continue
+                else:
+                    locus, alleles, vrs_json = line
+                    vrs_variant = json.loads(vrs_json)
+                    vrs_variant = gks_compute_seqloc_digest(vrs_variant)
+                    # serialize outputs to JSON and write to TSV
+                    vrs_json = json.dumps(vrs_variant)
+                    alleles = json.dumps(json.loads(alleles))
+                    f_out.write("\t".join([locus, alleles, vrs_json]))
+                    f_out.write("\n")
+                    counter += 1
+    end = timer()
+    logger.info(
+        "Computed %s SequenceLocation digests in %s seconds", counter, (end - start)
+    )
+    logger.info("Importing VRS records with computed SequenceLocation digests")
+    ht_with_location = hl.import_table(
+        computed_tmpfile, types={"locus": "tstr", "alleles": "tstr", "vrs_json": "tstr"}
+    )
+    ht_with_location_parsed = ht_with_location.annotate(
+        locus=hl.locus(
+            contig=ht_with_location.locus.split(":")[0],
+            pos=hl.int32(ht_with_location.locus.split(":")[1]),
+            reference_genome="GRCh38",
+        ),
+        alleles=hl.parse_json(ht_with_location.alleles, dtype=hl.tarray(hl.tstr)),
+    ).key_by("locus", "alleles")
+
+    return ht.drop("vrs_json").join(ht_with_location_parsed, how="left")
+
+
+def add_gks_vrs(ht: hl.Table):
+    """
+    Add GKS VRS variant annotation to a hail table.
+
+    Annotates ht with GA4GH GKS VRS structure, except for the variant.location._id,
```

Comment:
why can't location id be added with what was used previously:

Reply:
@theferrit32

Reply:
The location id can't be added here because these are added has hail expressions and they can't call into normal python functions like the SequenceLocation init and `ga4gh_identify`. That step is deferred to after the rows are `collect()`ed. The code flow is a little confusing, and because of this complexity around not being able to compute the location id in a hail expression, which I was not anticipating when I first started. Because of that, it might be better to do all the vrs/va stuff after the `collect()` call. The filtering done in `gnomad_gks_batch` is only based on the locus, so it doesn't need the vrs/va stuff to be in the table itself. 

@klaricch @ch-kr Maybe we can discuss this next week.